### PR TITLE
Pass vars between commands

### PR DESCRIPTION
**Pull request by AtebMT**
Commands should have access to the result of the previous commands that have a register option. This allows passing of information from command to command.

### CODE REVIEW

File: src/modules/task-runner.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+'use strict';
```
Comment (sashman) [inline src/modules/task-runner.js:1]: File name is in kebab case, is that intentional?

File: src/modules/task-runner.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    logger.info(`Executing command : ${parsedCommand.join(" ")}`);
+
+    const result = spawnSync(parsedCommand[0], _.tail(parsedCommand), spawnOptions);
+
+    if (command.register) {
+      this.vars[command.register] = result.stdout;
+    }
+
+    return (result.status !== 0 && !command.ignore_errors) ? false : true;
```
Comment (sashman) [inline src/modules/task-runner.js]: Use `!` at the start, rather than a tertiary

Comment (SlowShip) [inline src/modules/task-runner.js]: Could be 

```
return command.ignore_errors || result.status === 0
```

File: src/modules/task-runner.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  expandTokens(command) {
+    const template = _.template(command);
+
+    return template(this.vars);
+  }
+
+  resolveKeyValuePairs(array) {
+    const templated = this.vars ? _.map(array, a => this.expandTokens(a, this.vars)) : array;
```
Comment (sashman) [inline src/modules/task-runner.js]: Could extract `_.map(array, a => this.expandTokens(a, this.vars))` to a variable to make it easier to understand.

File: src/modules/task-runner.js (hunk 1 above)
Comment (AtebMT) [inline src/modules/task-runner.js:1]: Yes!

**Comment by SlowShip** [comment]
:+1:

**Pull request merged**